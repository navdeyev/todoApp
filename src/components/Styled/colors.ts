export enum colors {
    BLACK = '#000000',
    MINE_SHAFT = '#222222',
    DOVE_GREY = '#666666',
    DUSTY_GREY = '#999999',
    ALTO = '#dddddd',
    WHITE = '#ffffff'
}
