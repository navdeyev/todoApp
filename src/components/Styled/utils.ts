import {css, SimpleInterpolation} from 'styled-components';

export const rem = (px: number): string => {
    return `${px / 16}rem`;
};

enum MediaTypes {
    MOBILE = 'only screen and (max-width: 600px)',
    TABLET = 'only screen and (min-width: 601px) and (max-width: 1023px)',
    DESKTOP = 'only screen and (min-width: 1024px)'
}

const createMediaWrapper = (mediaType: MediaTypes) =>
    (strings: TemplateStringsArray, ...interpolations: SimpleInterpolation[]) =>
        css`@media ${mediaType} { ${ css(strings, ...interpolations) } }`;

export const media = {
    mobile: createMediaWrapper(MediaTypes.MOBILE),
    tablet: createMediaWrapper(MediaTypes.TABLET),
    desktop: createMediaWrapper(MediaTypes.DESKTOP)
};