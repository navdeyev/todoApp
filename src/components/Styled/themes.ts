import {colors} from './colors';

export interface ITheme {
    pageHeaderBgColor: string,
    pageTitleColor: string
}

export const defaultTheme: ITheme = {
    pageHeaderBgColor: colors.MINE_SHAFT,
    pageTitleColor: colors.WHITE
};

export const darkTheme: ITheme = {
    pageHeaderBgColor: colors.ALTO,
    pageTitleColor: colors.BLACK
};