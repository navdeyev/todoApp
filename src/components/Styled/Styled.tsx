import styled from 'styled-components';
import {media, rem} from './utils';

export const Heading1 = styled.h1`
    margin: 0;
    padding:0;
    font-weight: 700;
    font-size: ${ rem(24) };
    line-height: ${ rem(32) };
    
    ${ media.tablet`
        font-size: ${ rem(28) };
        line-height: ${ rem(36) };
    `}

    ${ media.desktop`
        font-size: ${ rem(32) };
        line-height: ${ rem(40) };
    `}
`;

export const Heading2 = styled.h2`
    margin: 0;
    padding:0;
    font-weight: 400;
    font-size: ${ rem(20) };
    line-height: ${ rem(28) };
    
    ${ media.tablet`
        font-size: ${ rem(22) };
        line-height: ${ rem(30) };
    `}

    ${ media.desktop`
        font-size: ${ rem(24) };
        line-height: ${ rem(32) };
    `}
`;

export const Paragraph = styled.p`
    margin: 0;
    padding:0;
    font-weight: 100;
    font-size: ${ rem(12) };
    line-height: ${ rem(16) };
    
    ${ media.tablet`
        font-size: ${ rem(14) };
        line-height: ${ rem(16) };
    `}
    
    ${ media.desktop`
        font-size: ${ rem(16) };
        line-height: ${ rem(18) };
    `}
`;