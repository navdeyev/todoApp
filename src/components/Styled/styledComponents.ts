import * as styledComponents from 'styled-components';
import {ThemedStyledComponentsModule} from 'styled-components';

import {ITheme} from './themes';

const {
    default: styled,
    css,
    injectGlobal,
    keyframes,
    ThemeProvider
} = styledComponents as ThemedStyledComponentsModule<ITheme>;

export {css, injectGlobal, keyframes, ThemeProvider};
export default styled;