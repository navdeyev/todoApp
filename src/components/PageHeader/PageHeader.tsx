import * as React from 'react';

import styled from 'components/Styled/styledComponents';
import {defaultTheme, ITheme} from 'components/Styled/themes';
import {rem} from 'components/Styled/utils';

export interface IPageHeaderProps {
    theme?: ITheme,
    className?: string,
    children: React.ReactNode
}

const Header: React.SFC<IPageHeaderProps> = props =>
    <header className={props.className}>{props.children}</header>;

const PageHeader = styled(Header)`
    display: flex;
    align-items: center;
    padding: ${ rem(20) };
    flex: 0 0 ${ rem(40) };
    background-color: ${ props => props.theme.pageHeaderBgColor };
`;

PageHeader.defaultProps = {theme: defaultTheme};
PageHeader.displayName = 'PageHeader';

export default PageHeader;