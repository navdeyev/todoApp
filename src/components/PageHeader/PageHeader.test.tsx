import {shallow} from 'enzyme';
import * as React from 'react';

import {IHasTheme} from 'components/Styled/themes';
import PageHeader from './PageHeader';

describe('PageHeader', () => {
    it('renders a themed PageHeader component', () => {
        const props: IHasTheme = {
            children: <div>Some text here</div>
        };
        const render = shallow(<PageHeader {...props}/>);
        expect(render.getElement()).toMatchSnapshot();

        const diverRender = shallow(<PageHeader {...props}/>).dive();
        expect(diverRender.getElement()).toMatchSnapshot();
    });
});
