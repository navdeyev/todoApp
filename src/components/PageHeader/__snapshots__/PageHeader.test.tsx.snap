// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader renders a themed PageHeader component 1`] = `
<Header
  className="sc-bdVaJa iwfSKB"
  theme={
    Object {
      "pageHeaderBgColor": "#222222",
      "pageTitleColor": "#ffffff",
    }
  }
>
  <div>
    Some text here
  </div>
</Header>
`;
